app/assets/javascripts/darkswarm/controllers/registration/registration_form_controller.js.coffee

```diff
@@ -19,9 +18,3 @@ angular.module('Darkswarm').controller "RegistrationFormCtrl", ($scope, Registra
 
   $scope.selectIfValid = (nextStep, form) ->
     RegistrationService.select(nextStep) if $scope.valid(form)
-
-  $scope.disableButton = ->
-    $scope.isDisabled = true
-
-  $scope.enableButton = ->
-    $scope.isDisabled = false
```

Comment:
(I can't comment on the related lines directly, so I commenting here) 
The actual issue is around the call to `EnterpriseRegistrationService` , the code here expect  `EnterpriseRegistrationService` to return a promise, but it doesn't. That said, it accepts a callback, so you could just pass it `$scope.enableButton` as a parameter and it should fix the problem.
see : https://github.com/openfoodfoundation/openfoodnetwork/blob/dab595f17f3df5a206639aa666b3440447a83fb7/app/assets/javascripts/darkswarm/services/enterprise_registration_service.js.coffee#L17-L40

```suggestion
 $scope.create = (form) ->
    if ($scope.valid(form)) 
      $scope.disableButton()
      EnterpriseRegistrationService.create($scope.enableButton)
```

Reply:
Aaah, very nice !! I was a bit stuck in my head to force user to go to stage 1, but it is faster to do this way.